maintenance_timesheet/models/maintenance_request.py

```diff
@@ -0,0 +1,57 @@
+# © 2019 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class MaintenanceRequest(models.Model):
+    _inherit = "maintenance.request"
+
+    timesheet_ids = fields.One2many(
+        string="Timesheets",
+        comodel_name="account.analytic.line",
+        inverse_name="maintenance_request_id",
+    )
+    timesheet_total_hours = fields.Float(compute="_compute_timesheet_total_hours")
+
+    def _add_followers(self):
+        """
+        Members of maintenance team are included as followers to automatically
+        grant request visibility and timesheet permissions for this request
+        """
+        super()._add_followers()
+        for request in self:
+            partner_ids = request.maintenance_team_id.member_ids.mapped(
+                "partner_id"
+            ).ids
+            request.message_subscribe(partner_ids=partner_ids)
+
+    @api.depends("timesheet_ids.unit_amount")
+    def _compute_timesheet_total_hours(self):
+        for request in self:
+            request.timesheet_total_hours = sum(
+                request.timesheet_ids.mapped("unit_amount")
+            )
+
+    def action_view_timesheet_ids(self):
+        """
+        Access to the current timesheets for this maintenance request
+        The view will be restricted to the current request and only HR managers
+        could create timesheets for every employee
+        """
+        self.ensure_one()
+        action = (
+            self.env.ref("maintenance_timesheet.timesheet_action_from_request")
+            .with_user(self.env.ref("base.user_admin"))
+            .read()[0]
```

Comment:
@etobella I've found that this change seems to work:

```suggestion
            self.env["ir.actions.act_window"]._for_xml_id("maintenance_timesheet.timesheet_action_from_request")
```

I've checked https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0 and this possible migration task is not listed. @pedrobaeza is that change correct when migrating from to v14?

Reply:
> 
> 
> @etobella I've found that this change seems to work:
> 
> I've checked https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0 and this possible migration task is not listed. @pedrobaeza is that change correct when migrating to v14?

@pedrobaeza @etobella could you check this? If the proposed approach is the right one, I'll update the PR. Thanks!

Reply:
I don't see this as a general change for 14.0, but specific business logic one, or am I missing something?

Reply:
I've found that demo user with his default permissions can't apparently access` ir.actions.act_window` model in readonly mode, and that didn't happen in v13, as far as I know. So, raw accesses (such as `read` method in order to change domain or context of an action) to this model will fail. Then, IMO during an addon migration I think this kind of code could be present sometimes, and I asked for the best practice you should recommend.

Anyway if `_for_xml_id()` does the trick, there's no problem in this case for me, and I'll use it.